src/Engine/Managers/EntityManager/EntityManager.cpp

```diff
@@ -79,13 +86,11 @@ Entity* EntityManager::getEntity( Core::Index idx ) const {
 
 std::vector<Entity*> EntityManager::getEntities() const {
     std::vector<Entity*> entities;
-    uint size = m_entities.size();
-
-    entities.reserve( size );
+    entities.reserve( m_entities.size() );
 
-    for ( uint i = 0; i < size; ++i )
+    for ( const auto& e : m_entities )
     {
-        entities.push_back( m_entities[i].get() );
+        entities.push_back( e.get() );

```

Comment:
Considering using `std::copy` (see https://github.com/STORM-IRIT/Radium-Engine/pull/309#pullrequestreview-120221716)

Reply:
using `std::copy` here is not possible since m_entities contains `std::shared_ptr<Entity>` while entities contains `Entity*`

Reply:
`std::transform` should do the job then.